src/common/Header.js

```diff
@@ -14,14 +17,25 @@ const StyledToolbar = styled(Toolbar)(({ theme }) => ({
   backgroundColor: '#FAFAFA',
 }));
 
-export default function Header(props) {
+export default function Header() {
+  const location = useLocation();
+  const isAdmin = location.pathname.includes('admin');
+  const isPatient = location.pathname.includes('patient');
+  const { data: patientData, isFetching: patientFetching } = useGetPatientInfoQuery();
+  const { data: adminData, isFetching: adminFetching } = useGetAdminInfoQuery();
+
+  useEffect(() => {
+    // do nothing
+  }, [patientFetching, patientData, adminData, adminFetching]);
+
   return (
     <Box sx={{ flexGrow: 1 }}>
       <AppBar position="fixed" elevation={0}>
         <StyledToolbar>
           <LogoIcon />
           <Box sx={{ flexGrow: 1 }} />
-          <LogoutButton />
+          {(isAdmin && adminData && <UserMenu name={adminData.name} isAdmin />)
+          || (isPatient && patientData && <UserMenu name={patientData.name} isPatient />)}
```

Comment:
Why not leave out the isPatient prop entirely? That would make isAdmin undefined here and still work in UserMenu, especially since this prop is never explicitly used.

Reply:
hmm not sure I follow